From 3e6e5ca744e8a3ae7ababb8eeec2cdf39d225002 Mon Sep 17 00:00:00 2001
From: Rene Rivera <grafikrobot@gmail.com>
Date: Fri, 30 Jan 2026 08:40:16 -0600
Subject: [PATCH 1/6] Fix `--abbreviate-paths` eating following arg.

The `--abbreviate-paths` option was not set as a flag. Which meant it incorrectly consumed ny following arg. Fixes #536
---
 doc/src/history.adoc   | 2 ++
 src/build/property.jam | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/tools/build/doc/src/history.adoc b/tools/build/doc/src/history.adoc
index 0d57cbc..8da345d 100644
--- a/tools/build/doc/src/history.adoc
+++ b/tools/build/doc/src/history.adoc
@@ -18,6 +18,8 @@
   -- _Paolo Pastori_
 * Fix argument count mismatch in Jam stack cleanup functions.
   -- _René Ferdinand Rivera Morell_
+* Fix `--abbreviate-paths` eating following arg.
+  -- _René Ferdinand Rivera Morell_
 
 == Version 5.4.2
 
diff --git a/tools/build/src/build/property.jam b/tools/build/src/build/property.jam
index cbbde2a..5451e0d 100644
--- a/tools/build/src/build/property.jam
+++ b/tools/build/src/build/property.jam
@@ -282,7 +282,8 @@ local rule identity ( string )
 
 
 args.add-arg abbreviate-paths : --abbreviate-paths
-    : "Compresses target paths by abbreviating each component." ;
+    : "Compresses target paths by abbreviating each component."
+    : flag ;
 if [ args.get-arg abbreviate-paths ]
 {
     .abbrev = abbreviate-dashed ;
-- 
2.51.0

